### DISCUSSION TITLE
Custom Sidebar Extension

### DISCUSSION THREAD

**Discussion by FreshlyBrewedCode** [directus/directus#10950] [Feature Requests]
Hi, first of all, thanks for this great project!
I could not find a discussion dedicated to this exact idea/feature so I am starting one here. 

#### Feature Description
It would be great if there was a way to add new modules/components to the sidebar using extensions. Currently, the main purpose of the sidebar is to expose additional context/view-specific metadata and tooling (e.g. revisions, comments, shares in the details view). Additionally, there is also a sticky section at the bottom (currently only contains the activity log) that shows the same elements regardless of the current context/view. I believe it would be great if an extension could add custom modules/components to both of these sections (1. contextual section, 2. sticky section).
![image](https://user-images.githubusercontent.com/11772241/148759227-da568926-bc53-40e0-a129-6ca45f102430.png)

For the contextual section, the extension would have to define a condition that determines if the sidebar component should be visible (e.g. only show the component for specific collections, layouts, etc.).

The custom components (both contextual and sticky) should also have access to the current context (e.g. which collection/item is currently displayed)

#### Example Use Cases
- Custom import/export/share functionality
  - Generate some custom shareable URLs
  - Send/Publish content to some third party tool/app
- Manage custom backend tasks (e.g. tasks triggered by a directus hook, like generating content thumbnails, etc.)
  - Show currently running tasks (sticky, or contextual for specific content)

#### Alternatives
It would be possible to create custom layouts or modules to implement some of the examples above. However, using the sidebar would be a more elegant approach in my opinion. Custom contextual components would also provide an easy way to extend the functionality of the existing layouts/views and extensions would not have to recreate existing functionality in custom layouts/modules (e.g. displaying a collection and allowing multi-selection to perform some batch operation).

**Comment by rijkvanzanten** [comment]
Like it! I've been thinking through a more "slot-based" extension type, that basically allows you to inject a custom vue component in a number of different "slots" in the app, like the sidebars, headers, etc (cc @nickrum).

**Reply by benhaynes** [reply]
Yeah, and I'd also like to mention that we've intended for these components to be extensions at some point... just not enough time to make it happen! Would love to really flesh this out and formalize how it will work in an RFC process. Seems really powerful for registering new custom functionality! ❤️

**Reply by snorbik** [reply]
@benhaynes Is there any update regarding this functionality?

**Reply by benhaynes** [reply]
No, not yet. But we're driving into improving the dev experience of extensions in a few weeks.

**Reply by ahmtaras** [reply]
Is there any development on this topic? Just what i need 🎉

**Reply by rijkvanzanten** [reply]
@ahmtaras Not yet! We're prioritizing based on community interest, and this is (unfortunately) not that often requested (as compared to other things)

**Reply by bartrail** [reply]
repeating the question 2 1/2 years later - is there any way to achieve this?

**Comment by SergeyRe** [comment]
i switched to directus from contentful. The feature discussed is presented there from the beginning. As for me -it was the most demanded. 
Please take a look to [Entry Sidebar App location](https://www.contentful.com/developers/docs/extensibility/app-framework/locations/#entry-sidebar) .

![image](https://www.contentful.com/developers/_assets/images/apps-locations-entry-sidebar.a7e6bfa56f.png)

> Sometimes the functionality you are hoping to achieve pertains more to the entry as a whole rather than an individual field. For example, imagine you want to show users the status of translations for the current entry in a third-party service. The sidebar location is perfect for this case because it provides users with important information about the entry they are currently viewing without obstructing the editorial experience.

**Comment by rijkvanzanten** [comment]
Heya! Thanks for opening this feature request!

This feature request has received over **15 votes** from the community. This means we'll move this feature request to the Under Review state! 

The Core team will schedule a meeting to review this request as soon as possible. The discussion will then be approved or denied. You may or may not be invited to join this meeting with the core team.

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._

**Reply by CaptnBook4Git** [reply]
Hey, it's been a while. What's the outcome here?

**Reply by niklasthunviktui** [reply]
I'm also very interested in this.

**Reply by bartrail** [reply]
Me too. Where can I leave an upvote? :D

**Reply by daibinhua888** [reply]
Me too. I'm also very interested in this.

**Reply by yoanntalagrand** [reply]
Highly interested as well

**Reply by jcomack** [reply]
Curious to hear if there's any progress on this front as I'm currently facing a situation where creating a completely new layout just to add  a button to the collection / item sidebar feels like overkill.

**Reply by JamesW1** [reply]
Hi there! This is not currently being actively worked on. We are in the process of refactoring the sidebar to support a cleaner content editing/publishing experience. 

Always curious to hear more about your specific use case? It might be something we can revisit once the above is finalised.
